dandi/metadata.py

```diff
@@ -339,6 +344,27 @@ def nwb2asset(nwb_path, digest=None, digest_type=None):
     return asset
 
 
+def get_default_metadata(path, digest=None, digest_type=None):
+    if digest is not None:
+        digest_model = models.Digest(
+            value=digest,
+            cryptoType=models.DigestType[digest_type],
+        )
+    else:
+        digest_model = None
+    dateModified = get_utcnow_datetime()
+    blobDateModified = ensure_datetime(os.stat(path).st_mtime)
+    if blobDateModified > dateModified:
+        lgr.warning("mtime of %s is in the future", path)
+    return models.BareAssetMeta.unvalidated(
+        contentSize=os.path.getsize(path),
+        digest=digest_model,
+        dateModified=dateModified,
+        blobDateModified=blobDateModified,
+        # encodingFormat # TODO
+    )
```

Comment:
we have significant code overlap between `get_default_metadata` and `nwb2asset`.  Could `nwb2asset` use `get_default_metadata` and add extra metadata?

BTW `wasGeneratedBy` pertinent and should be populated for any (nwb or non nwb) asset

Please either do above in this PR, or file a dedicated issue so we do not forget about these two.

Although we would not be able to go all the way with type annotations and validate with mypy, may be it is worth start adding it to functions we touch (even if without validation)?

Reply:
>  Could nwb2asset use get_default_metadata and add extra metadata?

Not without nontrivial rewrites.

Reply:
I've added `wasGeneratedBy` to the default metadata.

Reply:
Issue created for the first task: https://github.com/dandi/dandi-cli/issues/456

Reply:
thank you!